### PR TITLE
[Toolchain] Update deps

### PR DESCRIPTION
**Pull request by damassi**
Update Babel beta to 7 proper via `npx babel-upgrade --write`, and 

- `@types/react": "^16.4.13`
- `@types/styled-components": "^3.0.0`
- `@types/styled-system": "^3.0.3`
- `react": "^16.5.0`
- `react-dom": "^16.5.0`
- `styled-components": "^3.4.5`
- `styled-system: 3.0.3`

Corresponding PR: https://github.com/artsy/palette/pull/78

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit 8de4ca297e3125a55e1bf62baacfe09e3d0b96e4

https://deploy-preview-1227--artsy-reaction.netlify.com

File: src/Components/Forms/OrderForm/App.tsx [hunk 1]
```diff
@@ -60,7 +60,7 @@ const StepContainer = styled.div`
   margin: 0 auto;
 `
 
-export const StyledTitle = Title.extend`
+export const StyledTitle = styled(Title)`
```
Comment (damassi) [inline src/Components/Forms/OrderForm/App.tsx]: Each of these `styled(...)` replacements is due to `styled-components` removing this api method in version 4 with the warning failing our tests.

File: src/Styleguide/Elements/Flex.tsx [hunk 2]
```diff
@@ -36,7 +36,7 @@ import { ClassAttributes, HTMLAttributes } from "react"
 
 const flexGrow = style({
   prop: "flexGrow",
-  numberToPx: false,
+  // numberToPx: false,
```
Comment (ds300) [inline src/Styleguide/Elements/Flex.tsx]: You can just delete this. I think it was false by default even before 3.0 anyway 🤷‍♂️

**Comment by ds300** [comment]
Seems fine to me. Not to state the obvious, but force also needs a react update in order to continue to working with these changes.

**Comment by alloy** [comment]
Nice one, @damassi 👌 Just a few tiny questions

File: package.json [hunk 3]
```diff
@@ -88,10 +89,10 @@
     "@types/react-transition-group": "^2.0.7",
     "@types/relay-runtime": "^1.3.5",
     "@types/storybook__react": "^3.0.5",
-    "@types/styled-system": "^2.3.0",
+    "@types/styled-system": "^3.0.3",
     "babel-core": "7.0.0-bridge.0",
```
[the remark this answers is not in the document]
Comment (damassi) [reply package.json]: Unfortunately yes, somewhere in Jest this is still needed: 

```
Requires Babel "^7.0.0-0", but was loaded with "6.26.3". If you are sure you have a compatible version of @babel/core, it is likely that something in your build process is loading the wrong version. Inspect the stack trace of this error to look for the first entry that doesn't mention "@babel/core" or "babel-core" to see what is calling Babel.
```

File: package.json [hunk 4]
```diff
@@ -88,10 +89,10 @@
...
-    "babel-jest": "^23.4.0",
-    "babel-loader": "8.0.0-beta.0",
+    "babel-jest": "^23.4.2",
+    "babel-loader": "8.0.2",
```
[the remark this answers is not in the document]
Comment (damassi) [reply package.json]: Updated

**Pull request merged**